test/models/game_test.rb

```diff
@@ -11,8 +11,22 @@ class GameTest < ActiveSupport::TestCase
 
      assert_equal 32, g.pieces.count
      assert_equal "King", g.pieces.last.type
-     assert_equal 4, g.pieces.last.row_position
-     assert_equal 7, g.pieces.last.col_position
+     assert_equal 0, g.pieces.last.row_position
+     assert_equal 4, g.pieces.last.col_position
+   end
 
+   test 'two_queens' do
+    setup_game
+    # g.pieces.where({game_id:"14", type: "Queen"})
+    assert_equal(2, @game.pieces.where({game_id:"15", type: "Queen"}).count, failure_message = "kirubel where are you")
```

Comment:
Don't assume the game id will be a particular number.  In fact, when you call @game.pieces - all of the pieces returned will have the correct game_id already.  That's the beauty of the Rails associations.